docs/api-documentation-guide.md

```diff
@@ -0,0 +1,126 @@
+# The Standard Library API Documentation Guide
+
+_Authors_: @shafreenAnfar @daneshk @praneesha  
+_Reviewers_: @chamil321  
+_Created_: 2021/10/08  
+_Updated_: 2021/10/08
+
+## Overview
+
+API docs is another attribute that is authored, maintained and governed by the Standard Library team. This is the only official way in which the standard library team communicates about its module details in form of documentation. 
+
+The key purpose of this document is to help developers understand a given standard library module when needed in a quick and easy way. Therefore the structure of this document no is different from a well written research paper, book or newspaper article. In each case you start from top to bottom. At the top we have the synopsis whereas at the bottom we have all the fine-grained details. Basically, as you continue downwards, the details increase until you have all the information. The same applies for API docs as well but expressed in a different way.
+
+Following is the high level structure of API documentation.
+
+- **Topic** - In this its the module name
+- **Overview** - A high level description of the module which includes things such as concepts, philosophies, standards, practises, motives, etc which we followed when designing the module. Once read one should be able to understand the overall purpose of this module as a whole. Basically this includes the synopsis of the package. 
+- **Sub Topics** - Once understood the high level purpose of the module. If interested,  developers can further dig deeper into understanding modules functions, objects, records, etc.
+
+A good example for the above can be found in [1](https://pkg.go.dev/regexp#example_) and [2](https://docs.oracle.com/javase/7/docs/api/).
+
+## Materializing the Above
+
+In order to materialize the above we need make use of three things,
+
+-  Package.md documentation 
+   -  This is all about the distribution of the modules. Rendered in Ballerina central but not in API docs of ballerina.io 
+   -  This can include stuff like a banner saying this is owned by the standard library module, version compatibility, release dates, governing principles, security validations, etc.
```

Comment:
```suggestion
   -  This can include stuff like a banner saying this is owned by the standard library team, version compatibility, release dates, governing principles, security validations, etc.
```